src/main/kotlin/nl/info/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -441,12 +442,26 @@ class ProductaanvraagService @Inject constructor(
                     )
                 }
                 val firstZaakafhandelparameters = zaakafhandelparameters.first()
+                productaanvraag.betrokkenen?.forEach {
+                    it.inpBsn?.let {
```

Comment:
`let` and `if` can be combined using `takeIf` or `takeUnless` followed by `run`, ie.

```kotlin
it.inpBsn
    ?.takeUnless { firstZaakafhandelparameters.betrokkeneKoppelingen.brpKoppelen }
    .run { throw BetrokkeneNotAllowed() }
```

alternatively, just use a combined `if`, ie.

```kotlin
if ( it.inpBsn != null 
    && !firstZaakafhandelparameters.betrokkeneKoppelingen.brpKoppelen) {
    throw BetrokkeneNotAllowed()
}
```